src/services/XmppSocket.js

```diff
@@ -283,6 +283,25 @@ export default {
       this.context.$store.setContactPresence({ jid: fullJid.bare, presence: presence.show, status: presence.status })
     })
 
+    // listen for retracted messages
+    this.client.on('messageRetracted', (retracted) => {
+      const index = this.context.$store.messages.findIndex((message) => message.from.bare === retracted.from && message.stanzaId === retracted.stanzaId)
+      if (index === -1) {
+        // original message is not found (unknown or retracted sent by a third party)
+        return
```

Comment:
I added the control on the moderation message (`retracted.from` must be equal to the bare jid of the room `roomname@mucservice`) here